cli/bench_command.go

```diff
@@ -66,305 +62,540 @@ type benchCmd struct {
-	bench.Flag("multisubjectmax", "The maximum number of subjects to use in multi-subject mode (0 means no max)").Default("100000").IntVar(&c.multiSubjectMax)
-	bench.Flag("retries", "The maximum number of retries in JS operations").Default("3").IntVar(&c.retries)
-	bench.Flag("dedup", "Sets a message id in the header to use JS Publish de-duplication").Default("false").UnNegatableBoolVar(&c.deDuplication)
-	bench.Flag("dedupwindow", "Sets the duration of the stream's deduplication functionality").Default("2m").DurationVar(&c.deDuplicationWindow)
+
+	addCommonFlags := func(f *fisk.CmdClause) {
+		f.Flag("clients", "Number of concurrent clients").Default("1").IntVar(&c.numClients)
+		f.Flag("msgs", "Number of messages to publish or subscribe to").Default("100000").IntVar(&c.numMsg)
+		f.Flag("no-progress", "Disable progress bar").UnNegatableBoolVar(&c.noProgress)
+		f.Flag("csv", "Save benchmark data to CSV file").StringVar(&c.csvFile)
+		f.Flag("size", "Size of the test messages").Default("128").StringVar(&c.msgSizeString)
+		// TODO: support randomized payload data
+	}
+
+	addPubFlags := func(f *fisk.CmdClause) {
+		f.Flag("multisubject", "Multi-subject mode, each message is published on a subject that includes the publisher's message sequence number as a token").UnNegatableBoolVar(&c.multiSubject)
+		f.Flag("multisubjectmax", "The maximum number of subjects to use in multi-subject mode (0 means no max)").Default("100000").IntVar(&c.multiSubjectMax)
+		f.Flag("sleep", "Sleep for the specified interval after publishing each message").Default("0s").DurationVar(&c.sleep)
+	}
+
+	addJSCommonFlags := func(f *fisk.CmdClause) {
+		f.Flag("stream", "The name of the stream to create or use").Default(DefaultStreamName).StringVar(&c.streamName)
+		f.Flag("js-timeout", "Timeout for JS operations").Default("30s").DurationVar(&c.jsTimeout)
+	}
+
+	addJSPubFlags := func(f *fisk.CmdClause) {
+		f.Flag("create", "Create or update the stream first").UnNegatableBoolVar(&c.createStream)
+		f.Flag("purge", "Purge the stream before running").UnNegatableBoolVar(&c.purge)
+		f.Flag("storage", "JetStream storage (memory/file) for the \"benchstream\" stream").Default("file").EnumVar(&c.storage, "memory", "file")
+		f.Flag("replicas", "Number of stream replicas for the \"benchstream\" stream").Default("1").IntVar(&c.replicas)
+		f.Flag("maxbytes", "The maximum size of the stream or KV bucket in bytes").Default("1GB").StringVar(&c.streamMaxBytesString)
+		f.Flag("retries", "The maximum number of retries in JS operations").Default("3").IntVar(&c.retries)
+		f.Flag("dedup", "Sets a message id in the header to use JS Publish de-duplication").Default("false").UnNegatableBoolVar(&c.deDuplication)
+		f.Flag("dedupwindow", "Sets the duration of the stream's deduplication functionality").Default("2m").DurationVar(&c.deDuplicationWindow)
+	}
+
+	addKVFlags := func(f *fisk.CmdClause) {
+		f.Flag("purge", "Purge the stream before running").UnNegatableBoolVar(&c.purge)
```

Comment:
be good to use kv terminology here so bucket not stream etc

Reply:
Good catch, done.